### PR TITLE
Ucd 5 admin user approves application

### PR DESCRIPTION
**Pull request by jinnashravan**
We approve the application only when we satisfy below conditions

1. All review fields are set to APPROVE.
2. In notes and reason we should have a reason for decision.

### CODE REVIEW

File: request_a_govuk_domain/request/admin/model_admins.py [hunk 1]
```diff
@@ -375,7 +375,7 @@ def get_owner(self, obj):
 
     def response_change(self, request, obj):
         if "_approve" in request.POST:
-            if obj.is_approvable():
+            if self.model.is_approvable(obj):
```
Comment (dhvander) [inline request_a_govuk_domain/request/admin/model_admins.py:378]: Why was this changed?

File: request_a_govuk_domain/request/models/review.py [hunk 2]
```diff
@@ -18,91 +31,90 @@ class Review(models.Model):
...
 
     def is_approvable(self) -> bool:
-        return True
+        if (
+            self.registrar_details == RegistrarDetailsReviewChoices.APPROVE
+            and self.domain_name_availability
+            == DomainNameAvailabilityReviewChoices.APPROVE
+            and self.registrant_org == RegistrantOrgReviewChoices.APPROVE
+            and self.registrant_person == RegistrantPersonReviewChoices.APPROVE
+            and (
```
Comment (dhvander) [inline request_a_govuk_domain/request/models/review.py:104]: Do we need one more condition for the exemption file as we upload two files in some paths?

File: request_a_govuk_domain/request/migrations/0011_alter_application_status_and_more.py [hunk 3]
```diff
@@ -0,0 +1,407 @@
+# Generated by Django 4.2.13 on 2024-07-23 14:49
+
+import django.core.validators
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
```
Comment (dhvander) [inline request_a_govuk_domain/request/migrations/0011_alter_application_status_and_more.py:8]: Did we do that many changes to the model in this branch?

Comment (jinnashravan) [reply request_a_govuk_domain/request/migrations/0011_alter_application_status_and_more.py:8]: Yes

File: request_a_govuk_domain/request/admin/model_admins.py (hunk 1 above)
Comment (jinnashravan) [reply request_a_govuk_domain/request/admin/model_admins.py:378]: obj was Application, using self.model for Review model and passing in Application obj to is_approvable.

File: request_a_govuk_domain/request/models/review.py (hunk 2 above)
Comment (jinnashravan) [reply request_a_govuk_domain/request/models/review.py:100]: Done for the other paths now.

File: request_a_govuk_domain/request/admin/model_admins.py (hunk 1 above)
Comment (dhvander) [reply request_a_govuk_domain/request/admin/model_admins.py:378]: We may have to investigate this :)
This is the code block that calls this function, so obj is supposed to be review

    def save_model(self, request, obj, form, change):
        super().save_model(request, obj, form, change)
        # Save the application attributes
        if obj.application.status == ApplicationStatus.NEW:
            obj.application.status = ApplicationStatus.IN_PROGRESS
        # Change the owner to be the current user regardless if there is already a user
        # assigned or not
        obj.application.owner = request.user
        obj.application.save()

File: request_a_govuk_domain/request/models/review.py [hunk 4]
```diff
@@ -18,91 +31,101 @@ class Review(models.Model):
...
+            )
+            and (
+                not self.application.policy_exemption_evidence
+                or (
+                    self.application.policy_exemption_evidence
+                    and self.policy_exemption == PolicyExemptionReviewChoices.APPROVE
+                )
+            )
+            and (
+                not self.application.ministerial_request_evidence
```
Comment (dhvander) [inline request_a_govuk_domain/request/models/review.py:116]: I think we can take out the  self.application.ministerial_request_evidence from the or condition as it only reaches there if that is true? Same for the other two conditions we added above.

Comment (dhvander) [reply request_a_govuk_domain/request/models/review.py:116]: Sorry I didn't think about it earlier.

**Review by dhvander — APPROVED** [review]
